### PR TITLE
#3222 create api page

### LINKED ISSUE #2868: Move the “Collections” and “The can Package” sections off the home page
The “Collections” list is confusing (is “collections” a CanJS term?) and daunting (there are so many APIs!). Those sections might be more useful on a main “API” page, where someone could more easily find all of the CanJS APIs.

### LINKED ISSUE #3222: Create the API Docs page
This page should have _Core_, _Infrastructure_, _Ecosystem_, and _Legacy_ under it.

### PR DESCRIPTION
**Pull request by imaustink**
Added API Docs page according to the [design proposal](https://github.com/canjs/canjs/issues/2752#issuecomment-299532153).
screen shot 2017-05-11 at 5 09 09 pm

Remove "Collection" and "The Can Package" sections from home page.
__Bofore:__
image

__After:__
image

Closes #3222 
Closes #2868

### CODE REVIEW

**Comment by chasenlehara** [comment]
I forgot to assign this issue to you: https://github.com/canjs/canjs/issues/2868

I would make that change too since you added them to the API page. 😊

File: docs/can-canjs/can-api.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
+@page api API Docs
+@parent canjs 2
+@outline 0
+@package ../../package.json
+@templateRender <% %>
+@description
+Welcome to the CanJS docs!
```
Comment (chasenlehara) [inline docs/can-canjs/can-api.md]: This is shown when you hover over an item in the navigation, so I would add a little bit more text here about what’s in this section.

Comment (imaustink) [inline docs/can-canjs/can-api.md]: Okay, I've updated the description. Let me know if you don't like the new one. I can try to come up with alternatives.

**Comment by imaustink** [comment]
I was just gonna make a separate PR but I don't really see a point so I just updated this one to include https://github.com/canjs/canjs/issues/2868 :)

**Pull request merged**